packages/vue-router/src/createApp.ts

```diff
@@ -97,8 +99,20 @@ export const createApp = (
             ? [
                 router.beforeEach((to, from) => {
                   if (from !== START_LOCATION) {
+                    // check if the next path (to.path) is inside the mfe (in its routes)
+                    const isToAppRoute =
+                      routes.filter(
+                        (route) =>
+                          route.path.replace(/\/$/, "") ===
+                          to.path.replace(/\/$/, "")
+                      ).length > 0;
+
+                    const nextPathname = isToAppRoute
+                      ? [basename, to.path].join("/").replace(/\/{2,}/g, "/")
```

Comment:
we can use the `dedupeSlash` function here:
```
const nextPathname = isToAppRoute
    ? dedupeSlash([basename, to.path].join("/"))
    : to.path;
```

Reply:
ok, done in commit 9aa7d2f894286027b20bedba54c7ef10c4fdf54b